integration-cli/docker_cli_run_unix_test.go

```diff
@@ -1574,3 +1574,16 @@ func (s *DockerSuite) TestRunWithNanoCPUs(c *check.C) {
 	c.Assert(err, check.NotNil)
 	c.Assert(out, checker.Contains, "Conflicting options: Nano CPUs and CPU Period cannot both be set")
 }
+
+// #34542
+func (s *DockerSuite) TestRunVolumeLeakage(c *check.C) {
```

Comment:
I don't think you could test this with a unit test, since it requires starting containers in order to make sure that `-v /:/rootfs` won't cause unintended mount leakage.